yaramo/node.py

```diff
@@ -18,9 +19,17 @@ def __init__(self, **kwargs):
         self.connected_on_head = None
         self.connected_on_left = None
         self.connected_on_right = None
+        self.maximum_speed_on_left = None
+        self.maximum_speed_on_right = None
         self.connected_nodes: list['Node'] = []
         self.geo_node: GeoNode = None
 
+    def maximum_speed(self, node_a: 'Node', node_b: 'Node'):
```

Comment:
A name like `maximum_speed_between` (and maybe a comment) could make it clearer what this function does. Also, could we change the type hints from strings to the actual classes?